diff --git a/src/CLR/Core/CLR_RT_StackFrame.cpp b/src/CLR/Core/CLR_RT_StackFrame.cpp
index a93dc7e9b..7be9237ed 100644
--- a/src/CLR/Core/CLR_RT_StackFrame.cpp
+++ b/src/CLR/Core/CLR_RT_StackFrame.cpp
@@ -319,6 +319,7 @@ bool CLR_RT_StackFrame::PushInline(
 
     // increment the evalPos pointer so that we don't corrupt the real stack
     evalPos++;
+    evalPos[0].SetObjectReference(NULL);
     assm = calleeInst.m_assm;
     ip = ipTmp;
 
